File: easy/to_goat_latin.go
package easy

import (
	"strings"
)

func ToGoatLatin(sentence string) string {
	if len(sentence) == 0 {
		return ""
	}
	result := strings.Fields(sentence)
	for i, word := range strings.Fields(sentence) {
		suffix := strings.Repeat("a", i+1)
		if isVowel(word[0]) {
			result[i] = word + "ma" + suffix
		} else {
			result[i] = word[1:] + string(word[0]) + "ma" + suffix
		}
	}
	return strings.Join(result, " ")
}

func isVowel(ch byte) bool {
	return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' ||
		ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U'
}

File: easy/to_goat_latin.py
vowels = "aeiouAEIOU"


def to_goat_latin(sentence: str) -> str:
    words = []
    for i, word in enumerate(sentence.split(" ")):
        suffix = "a" * (i + 1)
        if word[0] in vowels:
            words.append(f"{word}ma{suffix}")
        elif word[0] not in vowels:
            words.append(f"{word[1::]}{word[0]}ma{suffix}")
    return " ".join(words)
